src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1284,15 +1288,15 @@ private ITryStatement CreateBoundTryStatementOperation(BoundTryStatement boundTr
 
         private ICatchClause CreateBoundCatchBlockOperation(BoundCatchBlock boundCatchBlock)
         {
-            Lazy<IBlockStatement> handler = new Lazy<IBlockStatement>(() => (IBlockStatement)Create(boundCatchBlock.Body));
-            ITypeSymbol caughtType = boundCatchBlock.ExceptionTypeOpt;
+            Lazy<IOperation> expressionDeclarationOrExpression = new Lazy<IOperation>(() => boundCatchBlock.ExceptionSourceOpt != null ? Create(boundCatchBlock.ExceptionSourceOpt) : null);
+            ITypeSymbol exceptionType = boundCatchBlock.ExceptionTypeOpt ?? ((CSharpCompilation)_semanticModel.Compilation).GetWellKnownType(WellKnownType.System_Exception);
```

Comment:
If `ExceptionTypeOpt` is `null`, we cannot assume the exception type is `Exception`.

Reply:
I should probably change this check to see if the catch `Declaration` is null, then default to `System.Exception`, otherwise use `ExceptionTypeOpt`

Reply:
If the declaration is missing, the type should be `null`.

Reply:
> If the declaration is missing, the type should be null.

I agree with this, It looks like C# emits a catch that can catch any type in this case.